src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1536,20 +1425,14 @@
    received-timestamp :- pls/Timestamp]
   (add-report!* report received-timestamp true))
 
-(def ^:dynamic *orphaned-path-gc-limit* 200)
-(def ^:dynamic *orphaned-value-gc-limit* 200)
-
 (defn garbage-collect!
   "Delete any lingering, unassociated data in the database"
   [db]
   (time!
    (:gc performance-metrics)
-   (jdbc/with-transacted-connection db
-     (delete-unassociated-params!)
-     (delete-unassociated-environments!))
-   ;; These require serializable because they make the decision to
-   ;; delete based on row counts in another table.
-   (jdbc/with-transacted-connection' db :serializable
-     (delete-orphaned-paths! *orphaned-path-gc-limit*))
-   (jdbc/with-transacted-connection' db :serializable
-     (delete-orphaned-values! *orphaned-value-gc-limit*))))
+   (do
+     (jdbc/with-transacted-connection db
+       (delete-unassociated-params!)
+       (delete-unassociated-environments!))
+     (jdbc/with-transacted-connection db
+       (delete-unused-fact-paths)))))
```

Comment:
Do these need to be in separate transactions for a reason?

Reply:
No, I put it in a separate transaction because it didn't need to be coordinated, and assumed that might reduce the potential for contention.